src/dbnode/storage/index/convert/convert.go

```diff
@@ -196,6 +196,14 @@ func FromSeriesIDAndEncodedTags(id ident.BytesID, encodedTags ts.EncodedTags) (d
 		byteOrder = serialize.ByteOrder
 		total     = len(encodedTags)
 	)
+	if total == 0 {
+		// No tags set for this series
+		return doc.Metadata{
+			ID:     clone(id.Bytes()),
```

Comment:
NIt: I believe it would be useful to check in the test that the new `id` is not backed by the same bytes (also adding the same coverage for non-empty case).

Reply:
https://github.com/m3db/m3/pull/3107/commits/5d40dbe5b7e6d79d8952de8f396c19880f1f466a